src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy

```diff
@@ -329,7 +329,8 @@ class ProtobufPlugin implements Plugin<Project> {
         task = project.tasks.create(extractProtosTaskName, ProtobufExtract) {
           description = "Extracts proto files/dependencies specified by 'protobuf' configuration"
           destDir = getExtractedProtosDir(sourceSetName) as File
-          inputs.files project.configurations[Utils.getConfigName(sourceSetName, 'protobuf')]
+          inputs.files(project.configurations[Utils.getConfigName(sourceSetName, 'protobuf')])
+                  .withPathSensitivity(PathSensitivity.RELATIVE)
```

Comment:
This is from configuration and presumably also need to be "NAME_ONLY"?

Reply:
Honestly, I'm not too familiar with the `protobuf` configuration. I don't think its a general statement that anything from a "cofiguration" should be name only.
The other `NAME_ONLY` in the `setupExtractIncludeProtosTask` was the classpath for the compile dependency which means essentially maven jars. These jars may be located in different local maven caches. 
On the other hand some configurations may have Kotlin files which require `RELATIVE` sensitivity but Java files require `NAME_ONLY` sensitivity because the java compiler ignores the files path and only looks at the package name.

Anyways, I changed this to be `NAME_ONLY` sensitivity. Let me know if there are any other changes I can make.

Reply:
I don't think I fully understand how the files from dependencies are considered by the cache. Take a jar file from `compile` configuration for example, is the jar as a single file considered as the input, or individual class files?

Reply:
Sounds reasonable. Do you know how the Gradle cache check if a directory (just a path, not a file tree) is up-to-date? We currently support such usage for the `protobuf` configuration (#294)

Reply:
I believe the PathSensitivity works in a similar way whether a FileTree or a Directory are passed to a tasks inputs. The PathSensitivity only applies to the path of the FileTree or the Directory. It doesn't change how the contents of either are cached.

Here is some more information on task relocatability:
https://guides.gradle.org/using-build-cache/#relocatability